src/chigame/users/views.py

```diff
@@ -59,13 +60,16 @@ def get_redirect_url(self):
 
 @login_required
 def user_list(request):
-    users = User.objects.all()
-    table = UserTable(users)
-    context = {"users": users, "table": table}
+    if request.user.is_staff:
+        users = User.objects.all()
+        table = UserTable(users)
+        context = {"users": users, "table": table}
 
-    # Add information about top ranking users, total points collected, etc.
+        # Add information about top ranking users, total points collected, etc.
 
-    return render(request, "users/user_list.html", context)
+        return render(request, "users/user_list.html", context)
+    else:
+        return HttpResponseNotFound("Access to link is restricted to admins")
```

Comment:
This is ok for now, but it's worth noting that this should redirect back to the user page or something instead.

Reply:
Got it, will including in closing statement!